tests/db/test_bigchain_api.py

```diff
@@ -232,236 +99,22 @@ def test_text_search(self, b, alice):
-        b.write_block(block)
-
-        response, status = b.get_transaction(tx.id, include_status=True)
-        # add validity information, which will be returned
-        assert tx.to_dict() == response.to_dict()
-        assert status == b.TX_UNDECIDED
-
-    @pytest.mark.usefixtures('inputs')
-    def test_read_transaction_invalid_block(self, b, user_pk, user_sk):
-        from bigchaindb.models import Transaction
-
-        input_tx = b.get_owned_ids(user_pk).pop()
-        input_tx = b.get_transaction(input_tx.txid)
-        inputs = input_tx.to_inputs()
-        tx = Transaction.transfer(inputs, [([user_pk], 1)],
-                                  asset_id=input_tx.id)
-        tx = tx.sign([user_sk])
-        # There's no need to b.write_transaction(tx) to the backlog
-
-        # create block
-        block = b.create_block([tx])
-        b.write_block(block)
-
-        # vote the block invalid
-        vote = b.vote(block.id, b.get_last_voted_block().id, False)
-        b.write_vote(vote)
-        response = b.get_transaction(tx.id)
-
-        # should be None, because invalid blocks are ignored
-        # and a copy of the tx is not in the backlog
-        assert response is None
-
-    @pytest.mark.usefixtures('inputs')
-    def test_read_transaction_invalid_block_and_backlog(self, b, user_pk, user_sk):
-        from bigchaindb.models import Transaction
-
-        input_tx = b.get_owned_ids(user_pk).pop()
-        input_tx = b.get_transaction(input_tx.txid)
-        inputs = input_tx.to_inputs()
-        tx = Transaction.transfer(inputs, [([user_pk], 1)],
-                                  asset_id=input_tx.id)
-        tx = tx.sign([user_sk])
-
-        # Make sure there's a copy of tx in the backlog
-        b.write_transaction(tx)
-
-        # create block
-        block = b.create_block([tx])
-        b.write_block(block)
-
-        # vote the block invalid
-        vote = b.vote(block.id, b.get_last_voted_block().id, False)
-        b.write_vote(vote)
-
-        # a copy of the tx is both in the backlog and in an invalid
-        # block, so get_transaction should return a transaction,
-        # and a status of TX_IN_BACKLOG
-        response, status = b.get_transaction(tx.id, include_status=True)
-        assert tx.to_dict() == response.to_dict()
-        assert status == b.TX_IN_BACKLOG
-
-    @pytest.mark.usefixtures('inputs')
-    def test_genesis_block(self, b):
-        from bigchaindb.backend import query
-        block = query.get_genesis_block(b.connection)
-
-        assert len(block['block']['transactions']) == 1
-        assert block['block']['transactions'][0]['operation'] == 'GENESIS'
-        assert block['block']['transactions'][0]['inputs'][0]['fulfills'] is None
-
-    @pytest.mark.genesis
-    def test_create_genesis_block_fails_if_table_not_empty(self, b):
-        from bigchaindb.common.exceptions import GenesisBlockAlreadyExistsError
-
-        with pytest.raises(GenesisBlockAlreadyExistsError):
```

Comment:
This exception can be removed from `bigchaindb/common/exceptions.py` now.

Reply:
Done